contracts/vaults/STETHVault/base/RibbonVault.sol

```diff
@@ -0,0 +1,682 @@
+
+    event Redeem(address indexed account, uint256 share, uint256 round);
+
+    event Withdraw(address account, uint256 amount, uint256 shares);
+
+    event CollectVaultFees(
+        uint256 performanceFee,
+        uint256 vaultFee,
+        uint256 round
+    );
+
+    /************************************************
+     *  CONSTRUCTOR & INITIALIZATION
+     ***********************************************/
+
+    /**
+     * @notice Initializes the contract with immutable variables
+     * @param _weth is the Wrapped Ether contract
+     * @param _usdc is the USDC contract
+     * @param _ldo is the LDO contract
+     * @param _gammaController is the contract address for opyn actions
+     * @param _marginPool is the contract address for providing collateral to opyn
+     * @param _gnosisEasyAuction is the contract address that facilitates gnosis auctions
+     * @param _crvPool is the steth/eth crv stables pool
+     */
+    constructor(
+        address _weth,
+        address _usdc,
+        address _ldo,
+        address _gammaController,
+        address _marginPool,
+        address _gnosisEasyAuction,
+        address _crvPool
+    ) {
+        require(_weth != address(0), "!_weth");
+        require(_usdc != address(0), "!_usdc");
+        require(_ldo != address(0), "!_ldo");
+
+        require(_gnosisEasyAuction != address(0), "!_gnosisEasyAuction");
+        require(_gammaController != address(0), "!_gammaController");
+        require(_marginPool != address(0), "!_marginPool");
+        require(_crvPool != address(0), "!_crvPool");
+
+        WETH = _weth;
+        USDC = _usdc;
+        LDO = _ldo;
+
+        GAMMA_CONTROLLER = _gammaController;
+        MARGIN_POOL = _marginPool;
+        GNOSIS_EASY_AUCTION = _gnosisEasyAuction;
+        STETH_ETH_CRV_POOL = _crvPool;
+    }
+
+    /**
+     * @notice Initializes the OptionVault contract with storage variables.
+     */
+    function baseInitialize(
+        address _owner,
+        address _keeper,
+        address _feeRecipient,
+        uint256 _managementFee,
+        uint256 _performanceFee,
+        string memory tokenName,
+        string memory tokenSymbol,
+        Vault.VaultParams calldata _vaultParams
+    ) internal initializer {
+        VaultLifecycleSTETH.verifyConstructorParams(
+            _owner,
+            _keeper,
+            _feeRecipient,
+            _performanceFee,
+            tokenName,
+            tokenSymbol,
+            _vaultParams
+        );
+
+        __ReentrancyGuard_init();
+        __ERC20_init(tokenName, tokenSymbol);
+        __Ownable_init();
+        transferOwnership(_owner);
+
+        keeper = _keeper;
+
+        feeRecipient = _feeRecipient;
+        performanceFee = _performanceFee;
+        managementFee = _managementFee.mul(10**6).div(WEEKS_PER_YEAR);
+        vaultParams = _vaultParams;
+
+        collateralToken = IWSTETH(0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0);
```

Comment:
hardcoding tad bit unprofessional. but best way without getting stack too deep error in initialize() and not adding wsteth as variable in contract (redundant to collateralToken)

Reply:
How would we be able to test this on Kovan if we hardcoded it?

Reply:
we can just create a kovan copy. its ugly but given constraints its best.